Fix(konnect): Konnect landing page, index, and menu

lena-larionova opened the pull request:
## Description

Big overhaul of the Konnect landing page and metadata based on [feedback from PM](https://docs.google.com/document/d/1x_jTzYc3wcfR8n0Is7cHgdn4HDXaaIZBii9xejnCAzI/edit?tab=t.0).

Main changes:
* `konnect-platform` as a "product" is now `konnect`. This is mainly a metadata change for purposes of the index page. This addresses feedback that users expect a `/konnect` index, and it doesn't align with the `/konnect` landing page.
  * I left the reference URLs as `konnect-platform`, since Konnect encompasses many things. While the landing page and index are `/konnect`, they both include references across all of the other products and components.
* Reworked the landing page to cover the "three layer cake", tried to simplify the cards, and deemphasized Gateway as the main/only product. Previously we were presenting the Gateway getting started guide as the only and recommended way in; now they can access the getting started guide for any piece they want.
* Added a button to take them to a Konnect free trial at the very top
* Added info about tooling
* Added Konnect to the top menu of the site

Fixes #2150 
Fixes #1924 - remove instead of replace. There's no clear marketing diagram, and the existing one is very outdated.

## Preview Links
Check the top menu under "Platform" for the Konnect link
https://deploy-preview-2665--kongdeveloper.netlify.app/konnect/
https://deploy-preview-2665--kongdeveloper.netlify.app/index/konnect/

cloudjumpercat commented on app/_landing_pages/konnect.yaml:
```diff
@@ -150,117 +213,74 @@ rows:
       - blocks: 
         - type: card
           config: 
             title: Customer-Managed Encryption Keys (CMEK)
```
I now see that the previous tile takes me to https://deploy-preview-2665--kongdeveloper.netlify.app/konnect-platform/authentication/ which contains links to Teams and roles. Maybe the wording in the Konnect authentication tile just needs to be revised so it's clear it will also contain info about teams/roles

cloudjumpercat commented on app/_landing_pages/konnect.yaml:
```diff
@@ -150,117 +213,74 @@ rows:
-              url: /mesh-manager/
-              align: end
+            blocks:
+            - type: unordered_list
```
I'd prefer these as tiles instead

cloudjumpercat commented on app/_landing_pages/konnect.yaml:
```diff
@@ -273,110 +293,55 @@ rows:
-  
+              products:
+                  - gateway
+              works_on:
```
Looks like this `works_on` isn't working as expected since it's directing me to /how-to/?products=gateway

lena-larionova replied:
Hmm. It is working as configured (gateway how-tos + compatible with konnect) but now that I think about it, having only "gateway" as a product doesn't make sense anymore.

lena-larionova commented:
@cloudjumpercat applied your feedback, here's how I approached it:
* Support & account tiles: turned them back into tiles but moved them to the bottom of the page. This way users can cmd + f to find the info but it's not overwhelming them with the amount of tiles up front
* fixed the works_on filter, and removed the product filter for how-tos. Now the link _only_ filters on "works on konnect", which is more accurate
* shortened the references list from 10 to 7, raised how-tos to 7 to match
* shortened the link descriptions on the runtimes tiles - they weren't fitting well before and were wrapping awkwardly. Let me know you think the "learn", "get started", and "manage" links work

Old:
<img width="1267" height="527" alt="Screenshot 2025-08-26 at 1 26 54 PM" src="https://github.com/user-attachments/assets/ca529c79-d603-4e51-a38f-072bd3bafa42" />


New:
<img width="1277" height="487" alt="Screenshot 2025-08-26 at 1 21 38 PM" src="https://github.com/user-attachments/assets/4768c65e-43c1-4dfe-ad4d-bd1831b3e5a5" />
